src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -228,6 +228,42 @@ class LiteralKeywordSegment(KeywordSegment):
     type = "literal"
 
 
+class BinaryOperatorSegment(CodeSegment):
+    """An binary operator segment.
+
+    Defined here for type inheritance.
+    """
+
+    type = "binary_operator"
+
+
+class CompositeBinaryOperatorSegment(BaseSegment):
+    """An composite binary operator segment.
+
+    Defined here for type inheritance.
+    """
+
+    type = "binary_operator"
+
+
+class ComparisonOperatorSegment(CodeSegment):
+    """An comparison operator segment.
+
+    Defined here for type inheritance.
+    """
+
+    type = "comparison_operator"
+
+
+class CompositeComparisonOperatorSegment(BaseSegment):
```

Comment:
What does the word "composite" mean here?

Reply:
Just that it's a operator made up of multiple raw segments. The reason for having two versions here is to have one which derives from `RawSegment` and one which derives from `BaseSegment`. There's probably scope in future for condensing these into two and rationalising the parse structure, but this is the closest to current live usage.